providers/linux/vmstat.go

```diff
@@ -0,0 +1,61 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package linux
+
+import (
+	"reflect"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/go-sysinfo/types"
+)
+
+// parseVMStat parses the contents of /proc/vmstat
+func parseVMStat(content []byte) (*types.VMStatInfo, error) {
+	vmStat := &types.VMStatInfo{}
```

Comment:
How about something like this where you bootstrap a mapping just once?

```
// vmstatTagToFieldIndex contains a mapping of json struct tags to struct field indices.
var vmstatTagToFieldIndex map[string]int

func init(){
	var vmstat VMStatInfo
	val := reflect.ValueOf(vmstat)
	typ := reflect.TypeOf(vmstat)

	vmstatTagToFieldIndex = map[string]int{}
	for i := 0; i < val.NumField(); i++ {
		field := typ.Field(i)
		if tag := field.Tag.Get("json"); tag != "" {
			vmstatTagToFieldIndex[tag] = i
		}
	}
}
```

```
	var vmStat VMStatInfo
	val := reflect.ValueOf(&vmStat).Elem()

	if idx, found := vmstatTagToFieldIndex["my_key"]; found {
		f := val.Field(idx)
		if f.CanSet() {
			f.SetUint(42)
		}
	}
```

Reply:
I briefly thought of that but decided against it, but it didn't occur to me to use the struct index instead of the raw Value. Let's try that.